### PR TITLE
Replace ioutil for Go1.16 update

### LINKED ISSUE #114: Replace ioutil usage
With Go 1.16, the `ioutil` package is being deprecated: https://www.srcbeat.com/2021/01/golang-ioutil-deprecated/. We should start replacing its usage.

### PR DESCRIPTION
**Pull request by u5surf**
## What does this PR do?

Fixes : #114 

## Why is it important?
Also #114 said,  Go 1.16, the ioutil package is being deprecated.

## Checklist

- [x] I have added test packages to [`code/go/internal/validator/test/packages`](https://github.com/elastic/package-spec/tree/master/code/go/internal/validator/test/packages) that prove my change is effective.

Nothing additions, but I confirmed the regression. That was fine at all.
```
u5surf@ik1-321-20921:~/package-spec go version
go version go1.16beta1 linux/amd64
u5surf@ik1-321-20921:~/package-spec/code/go make test
=== RUN   TestBundledSpecs
--- PASS: TestBundledSpecs (0.00s)
PASS
ok      github.com/elastic/package-spec/code/go/internal        0.005s
?       github.com/elastic/package-spec/code/go/internal/spec   [no test files]
=== RUN   TestValidationErrorsMultiple
--- PASS: TestValidationErrorsMultiple (0.00s)
=== RUN   TestValidationErrorsSingle
--- PASS: TestValidationErrorsSingle (0.00s)
=== RUN   TestValidationErrorsNone
--- PASS: TestValidationErrorsNone (0.00s)
=== RUN   TestNewPackage
=== RUN   TestNewPackage/non_existent
=== RUN   TestNewPackage/no_manifest
=== RUN   TestNewPackage/no_spec_version
=== RUN   TestNewPackage/good
--- PASS: TestNewPackage (0.00s)
    --- PASS: TestNewPackage/non_existent (0.00s)
    --- PASS: TestNewPackage/no_manifest (0.00s)
    --- PASS: TestNewPackage/no_spec_version (0.00s)
    --- PASS: TestNewPackage/good (0.00s)
=== RUN   TestNewSpec
--- PASS: TestNewSpec (0.00s)
PASS
ok      github.com/elastic/package-spec/code/go/internal/validator      0.013s
?       github.com/elastic/package-spec/code/go/internal/yamlschema     [no test files]
=== RUN   TestValidateFile
=== RUN   TestValidateFile/missing_pipeline_dashes
=== RUN   TestValidateFile/missing_image_files
=== RUN   TestValidateFile/good
=== RUN   TestValidateFile/deploy_docker
=== RUN   TestValidateFile/input_template
=== RUN   TestValidateFile/bad_deploy_variants
--- PASS: TestValidateFile (0.04s)
    --- PASS: TestValidateFile/missing_pipeline_dashes (0.01s)
    --- PASS: TestValidateFile/missing_image_files (0.01s)
    --- PASS: TestValidateFile/good (0.02s)
    --- PASS: TestValidateFile/deploy_docker (0.00s)
    --- PASS: TestValidateFile/input_template (0.00s)
    --- PASS: TestValidateFile/bad_deploy_variants (0.00s)
=== RUN   TestValidateItemNotAllowed
=== RUN   TestValidateItemNotAllowed/wrong_kibana_filename
--- PASS: TestValidateItemNotAllowed (0.00s)
    --- PASS: TestValidateItemNotAllowed/wrong_kibana_filename (0.00s)
PASS
ok      github.com/elastic/package-spec/code/go/pkg/validator   0.048s
u5surf@ik1-321-20921:~/package-spec/code/go
```

- [x] I have added an entry in [`versions/N/changelog.yml`](https://github.com/elastic/package-spec/blob/master/versions/1/changelog.yml).

I expect to be adding after once sending this PR
## Related issues

- Closes #114

### CODE REVIEW

**Comment by ycombinator** [comment]
Thanks for the contribution, @u5surf. I was on holiday last week but this PR is on my radar now. I plan to look at it in the next day or two. In the meantime, could you address the merge conflict please? Thanks!

File: versions/1/changelog.yml [hunk 1]
```diff
@@ -52,3 +52,6 @@
   - description: Add fields for matching datasets dynamically
     type: enhancement
     link: https://github.com/elastic/package-spec/pull/102
+  - description: Replace ioutil for Go1.16 update
+    type: breaking-change
```
Comment (ycombinator) [inline versions/1/changelog.yml:56]: I'm not sure this is a breaking change. No public APIs have changed in this PR. I would just make this `type: enhancement` for now.

**Comment by ycombinator** [comment]
Hi @u5surf, looks like there's a merge conflict. Please rebase this PR on `master` to resolve that conflict. Thanks!

**Comment by ycombinator** [comment]
jenkins, please test this.

**Comment by ycombinator** [comment]
jenkins, run the tests please

**Comment by ycombinator** [comment]
> looks like there's a merge conflict. Please rebase this PR on master to resolve that conflict. Thanks!

@u5surf, never mind, I have resolved the merge conflict for you.

**Comment by ycombinator** [comment]
jenkins, run the tests please

**Comment by ycombinator** [comment]
jenkins, run the tests please

**Comment by ycombinator** [comment]
jenkins, run the tests please

**Comment by ycombinator** [comment]
CI is failing with this error:

```
[2021-01-27T17:20:52.948Z] 	io/fs: package io/fs is not in GOROOT (/var/lib/jenkins/workspace/gest-manager_package-spec_PR-115/.gvm/versions/go1.15.6.linux.amd64/src/io/fs)
```

This makes sense since we're currently using Go 1.15.6 for this project and the `io/fs` package isn't available until Go 1.16. This PR is still valid; we will just have to keep it around until we update the project to Go 1.16. There is no specific date or timeline for this update yet. I will post something here when we've made the update.

**Comment by u5surf** [comment]
@ycombinator 
Indeed, there is not such a package yet. If you wanna check right now, can you use `go1.16beta1` ?https://golang.org/dl/#unstable

**Comment by ycombinator** [comment]
@u5surf Looks like there's an RC out for go1.16 now. I will try to test this PR locally with that but we will still need to wait for CI to go green before we can merge it. For that to happen, the Golang version in CI will need to be bumped up to go1.16. This will not happen right after go1.16 is released — we tend to wait for some time, usually at least until the first patch release (so go1.16.1), before we feel comfortable upgrading to the new minor.

**Comment by mtojek** [comment]
I'm afraid that this PR is outdated now. Closing it.

**Pull request closed without merging**